packages/bitcore-wallet-client/lib/verifier.js

```diff
@@ -180,11 +181,24 @@ Verifier.checkPaypro = function(txp, payproOpts) {
     amount = txp.amount;
   }
 
+
+  if (amount != payproOpts.amount)
+    return false;
+
+
+  if (txp.coin == 'btc' && toAddress != payproOpts.toAddress)
+    return false;
+
+  // To circunvent cashaddr/legacy address problems...
+  if (txp.coin == 'bch' && (new BCHAddress(toAddress).toString()) != (new BCHAddress(payproOpts.toAddress).toString())) 
```

Comment:
In the future we may want to extract out these two checks into a multi-currency address validation function
I'm imagining something like this
```
CurrencyAddress.isValid(currency, address)
```

Reply:
nice. In this case I need `address1.isEqual(address2)`. Do you think is a good idea to add this to our libraries?

Reply:
(the problem here is that string comparision  of `.toString()` does not always work, since older versions of bitcore-lib-cash will render the address add legacy string and newer as cashaddr.